app/main.py

```diff
@@ -1,6 +1,20 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    dict_of_functions = {}
+
+    @wraps(func)
+    def wrapper(*args: object, **kwargs: object) -> object:
+        key = (func.__name__, args, frozenset(kwargs.items()))
```

Comment:
Why do you need to store func.__name__ ? It is redundant for each function dict_of_functions will be created new